### PR TITLE
Use real data for nearby items

### LINKED ISSUE #35: Populate nearby items with real data
Both the list and the detail page

### PR DESCRIPTION
**Pull request by frcroth**
Closes #35

### CODE REVIEW

**Review by henrijaeger — APPROVED** [review]

File: app/src/main/java/app/wefridge/wefridge/EditFragment.kt [hunk 1]
```diff
@@ -474,17 +474,8 @@ class EditFragment : Fragment() {
...
-            Unit.KILOGRAM.value -> getString(R.string.item_unit_kilogram)
-            Unit.LITER.value -> getString(R.string.item_unit_liter)
-            Unit.MILLILITER.value -> getString(R.string.item_unit_milliliter)
-            Unit.OUNCE.value -> getString(R.string.item_unit_ounce)
-            Unit.PIECE.value -> getString(R.string.item_unit_piece)
-            else -> null
-        }
-
+    private fun matchUnitValueToUnitDropdownSelection(): String {
+        return model?.unit.toString()
```
Comment (PassiDel) [inline app/src/main/java/app/wefridge/wefridge/EditFragment.kt:478]:
```suggestion
        return (model?.unit ?: Unit.PIECE).display(requireContext())
```

File: app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt [hunk 2]
```diff
@@ -67,11 +67,11 @@ class NearbyDetailFragment : Fragment() {
         super.onViewCreated(view, savedInstanceState)
 
         // TODO: populate textviews with actual datamodel
-        binding.quantity.text = ""
+        binding.quantity.text = "${model.quantity} ${model.unit}"
```
Comment (PassiDel) [inline app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt:70]:
```suggestion
        binding.quantity.text = "${model.quantity} ${model.unit.display(requireContext())}"
```

File: app/src/main/java/app/wefridge/wefridge/model/Unit.kt [hunk 3]
```diff
@@ -13,4 +13,15 @@ enum class Unit(val value: Int) {
...
+    override fun toString(): String {
+        return when (this) {
+            GRAM -> "g"
+            KILOGRAM -> "kg"
+            LITER -> "l"
+            MILLILITER -> "ml"
+            OUNCE -> "oz"
+            PIECE -> "pcs"
+        }
+    }
```
Comment (PassiDel) [inline app/src/main/java/app/wefridge/wefridge/model/Unit.kt:26]:
```suggestion
    fun display(ctx: Context): String = ctx.getString(_display)
```

File: app/src/main/java/app/wefridge/wefridge/model/Unit.kt [hunk 4]
```diff
@@ -13,4 +13,15 @@ enum class Unit(val value: Int) {
         fun getByValue(value: Int?) = VALUES.firstOrNull { it.value == value }
 
     }
```
Comment (PassiDel) [inline app/src/main/java/app/wefridge/wefridge/model/Unit.kt:15]: Move display string to Enum constructor to keep a single source of truth.
```kt
enum class Unit(val value: Int, @StringRes val _display: Int) {
    GRAM(0, R.string.item_unit_gram),
    KILOGRAM(1, R.string.item_unit_kilogram),
    LITER(2, R.string.item_unit_liter),
    MILLILITER(3, R.string.item_unit_milliliter),
    OUNCE(4, R.string.item_unit_ounce),
    PIECE(5, R.string.item_unit_piece);

...
```

**Review by PassiDel — COMMENTED** [review]
Move display string to Enum constructor to keep a single source of truth.

**Review by AlexTerhar — APPROVED** [review]

**Review by PassiDel — APPROVED** [review]

**Pull request merged**